### ISSUE TITLE
Authentication tests

### ISSUE THREAD

**Issue by SamLubbers** [xrnl/aginun#146]
## Problem that this feature solves

Our authentication server is still not really stable. Sometimes we cannot connect to it or it takes too long to respond. As a result, we run the risk of deploying the application without users being able to log in.

In addition, we want to limit certain functionality only for logged in users. As we grow the functionality on this application it will become harder to track this manually and we might accidentally give not logged in users to perform functionality reserved for logged in users (for example, delete a role)

## Test cases

To start off, we need the following two tests for the authentication server:

* if we call the authentication server with correct credentials, we get the expected data and are logged in.
* if we call the authentication server with incorrect credentials, we get an error message and are not logged in.

In addition, we need unit tests for aspects of the front-end related to the authentication

* If the user is logged in, the following buttons should be rendered
  * logout (in `src/components/TheAppBar.vue`)
  * create new role (in `src/views/RolesOverview.vue`)
* for roles that they have created, all of these buttons should be rendered in `src/components/roles/RoleViewDialog.vue`
  * edit role
  * delete role
  * mark role as taken
* if the user is not logged in, none of the previously mentioned buttons should be rendered, and the logout button should be rendered in the navbar.

Finally, once #134 is merged, we should also add unit tests for that component. 

If you think that this issue is too big you can split it up into smaller more manageable issues :innocent:
